### PR TITLE
[2.0.1] Clear DbCommand.Parameters only after reader is disposed

### PR DESCRIPTION
**Pull request by smitpatel**
Porting #9277 as per #9309

### CODE REVIEW

File: src/EFCore.Relational/Storage/Internal/RelationalCommand.cs [hunk 1]
```diff
@@ -225,10 +225,9 @@ public class RelationalCommand : IRelationalCommand
             }
             finally
             {
-                dbCommand.Parameters.Clear();
```
Comment (ajcvickers) [inline src/EFCore.Relational/Storage/Internal/RelationalCommand.cs]: Quirk?

Comment (smitpatel) [reply src/EFCore.Relational/Storage/Internal/RelationalCommand.cs]: Did we decide to quirk this? I couldn't find note anywhere.

Comment (ajcvickers) [reply src/EFCore.Relational/Storage/Internal/RelationalCommand.cs]: We're quirking everything by default, only not quirking if there is a compelling reason not to.

**Comment by smitpatel** [comment]
@ajcvickers - updated.

**Pull request merged**